tests/e2e/raw_block_volume.go

```diff
@@ -596,4 +724,561 @@ var _ = ginkgo.Describe("raw block volume support", func() {
+
+		// Write and read some data on raw block volume inside the pod.
+		// Use same devicePath for raw block volume here as used inside podSpec by createPod().
+		// Refer setVolumes() for more information on naming of devicePath.
+		volumeIndex := 1
+		devicePath := fmt.Sprintf("%v%v", pod_devicePathPrefix, volumeIndex)
+		ginkgo.By(fmt.Sprintf("Write and read data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		verifyDataOnRawBlockVolume(namespace, pod.Name, devicePath, "This is write before expansion")
+
+		defer func() {
+			// Delete Pod.
+			ginkgo.By(fmt.Sprintf("Deleting the pod %s in namespace %s", pod.Name, namespace))
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			isDiskDetached, err := e2eVSphere.waitForVolumeDetachedFromNode(client,
+				pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(isDiskDetached).To(gomega.BeTrue(),
+				fmt.Sprintf("Volume %q is not detached from the node %q", pv.Spec.CSI.VolumeHandle, pod.Spec.NodeName))
+			if guestCluster {
+				ginkgo.By("Waiting for 30 seconds to allow CnsNodeVMAttachment controller to reconcile resource")
+				time.Sleep(waitTimeForCNSNodeVMAttachmentReconciler)
+				verifyCRDInSupervisorWithWait(ctx, f, pod.Spec.NodeName+"-"+svcPVCName,
+					crdCNSNodeVMAttachment, crdVersion, crdGroup, false)
+			}
+		}()
+
+		ginkgo.By("Increase PVC size and verify online volume resize")
+		increaseSizeOfPvcAttachedToPod(f, client, namespace, pvc, pod)
+
+		// Verify original data on raw block volume after expansion
+		ginkgo.By(fmt.Sprintf("Verify previously written data on raw volume attached to: %v at path %v", pod.Name,
+			pod.Spec.Containers[0].VolumeDevices[0].DevicePath))
+		readDataFromRawBlockVolume(namespace, pod.Name, devicePath, "This is write before expansion")
+	})
+
+	/*
+	   Test to verify volume expansion is supported if allowVolumeExpansion
+	   is true in StorageClass, PVC is created and offline and not attached
+	   to a Pod before the expansion.
+	   Steps
+	   1. Create StorageClass with allowVolumeExpansion set to true.
+	   2. Create raw block PVC which uses the StorageClass created in step 1.
+	   3. Wait for PV to be provisioned.
+	   4. Wait for PVC's status to become Bound.
+	   5. Create pod using PVC on specific node.
+	   6. Wait for Disk to be attached to the node.
+	   7. Write some data to raw block PVC.
+	   8. Detach the volume.
+	   9. Modify PVC's size to trigger offline volume expansion.
+	   10. Create pod again using PVC on specific node.
+	   11. Wait for Disk to be attached to the node.
+	   12. Verify data written on PVC before expansion.
+	   13. Delete pod and Wait for Volume Disk to be detached from the Node.
+	   14. Delete PVC, PV and Storage Class.
+	*/
+	ginkgo.It("[csi-block-vanilla] [csi-block-vanilla-parallelized] [csi-guest] "+
```

Comment:
guest cluster support was supposed to come later right?

Reply:
raw block  volumes should work as is with guest cluster, Supervisor support is not there at present. Hence added tc for guest as well, will be testing the tc on GC setup once available.

Reply:
Ran expansion tcs on guest cluster setup, where it passed successfully.
Please refer https://gist.github.com/akankshapanse/38c85dad8db3ff0a957454212b9c2b1c